### PR TITLE
feat(anta.cli): Add an --enable flag

### LINKED ISSUE #255: Better handle when cli drops the user to privileged mode
Example EOS config:
```bash
tacacs-server host 172.25.99.99 single-connection key 7 <blah>
tacacs-server host 172.22.99.99 single-connection key 7 <blah>
!
aaa group server tacacs+ tacacs-ln
   server 172.25.99.99
!
aaa group server tacacs+ tacacs-ny
   server 172.22.99.99
!
no aaa root
aaa group server tacacs+ tacacs-ln
aaa group server tacacs+ tacacs-ny
aaa authentication login default group tacacs-ny group tacacs-ln local
aaa authentication enable default group tacacs-ny group tacacs-ln local
aaa authorization serial-console
aaa authorization exec default group tacacs-ny group tacacs-ln local
aaa authorization commands all default group tacacs-ny group tacacs-ln local
aaa accounting exec default start-stop group tacacs-ny group tacacs-ln
aaa accounting commands all default start-stop group tacacs-ny group tacacs-ln
```

errors associated:
`CLI Exception: Internal error <class 'CliParserCommon.AuthzDeniedError'>: Authorization denied for command 'enable': Cannot authorize command for unknown sessionId 1933`

It seems an enable is being passed even if no enable password is set for the config of anta.

https://github.com/arista-netdevops-community/anta/blob/master/anta/device.py#L247

### PR DESCRIPTION
**Pull request by gmuloc**
**DO NOT MERGE BEFORE READY FOR 0.7.0**

In 0.6.0 by default we inject `enable` or `enable` + password if `--enable-password` is set.

In some environments, it is possible that `enable` is forbidden to the username running the test (via tacacs, locally, radius, ...)

This PR adds an enable flag. By default enable won't be injected anymore. To inject the enable-password, the enable flag MUST be set

```
Usage: anta [OPTIONS] COMMAND [ARGS]...

  Arista Network Test Automation (ANTA) CLI

Options:
  --version                       Show the version and exit.
  --username TEXT                 Username to connect to EOS  [env var:
                                  ANTA_USERNAME; required]
  --password TEXT                 Password to connect to EOS  [env var:
                                  ANTA_PASSWORD; required]
  --timeout INTEGER               Global connection timeout  [env var:
                                  ANTA_TIMEOUT; default: 5]
  --insecure                      Disable SSH Host Key validation  [env var:
                                  ANTA_INSECURE]
  --enable                        Add enable mode towards the devices if
                                  required to connect  [env var: ANTA_ENABLE]
  --enable-password TEXT          Enable password if required to connect,
                                  --enable MUST be set  [env var:
                                  ANTA_ENABLE_PASSWORD]
  -i, --inventory FILE            Path to the inventory YAML file  [env var:
                                  ANTA_INVENTORY; required]
  --log-level, --log [CRITICAL|ERROR|WARNING|INFO|DEBUG]
                                  ANTA logging level  [env var:
                                  ANTA_LOG_LEVEL; default: INFO]
  --ignore-status                 Always exit with success  [env var:
                                  ANTA_IGNORE_STATUS]
  --ignore-error                  Only report failures and not errors  [env
                                  var: ANTA_IGNORE_ERROR]
  --help                          Show this message and exit.

Commands:
  debug  Debug commands for building ANTA
  exec   Execute commands to inventory devices
  get    Get data from/to ANTA
  nrfu   Run NRFU against inventory devices
```

Fixes #255 

TODO: see to make this flag per-device rather than inventory wide.

### CODE REVIEW

File: anta/cli/__init__.py [hunk 1]
```diff
@@ -49,10 +49,18 @@
...
+    is_flag=True,
+    default=False,
+    help="Add enable mode towards the devices if required to connect",
+    show_default=True,
+)
 @click.option(
     "--enable-password",
     show_envvar=True,
-    help="Enable password if required to connect",
+    help="Enable password if required to connect, --enable MUST be set",
```
Comment (mtache) [inline anta/cli/__init__.py:63]: `--enable MUST be set`: this is not tested in the code.

File: anta/device.py [hunk 2]
```diff
@@ -244,14 +248,15 @@ async def collect(self, command: AntaCommand) -> None:
         """
         try:
             commands = []
-            if self._enable_password is not None:
+            if self.enable and self._enable_password is not None:
```
Comment (mtache) [inline anta/device.py:251]: So I can provide enable_password without enable and ANTA will silently ignore enable_password.

**Review by mtache — CHANGES_REQUESTED** [review]

File: anta/device.py (hunk 2 above)
Comment (gmuloc) [reply anta/device.py:251]: no more silence in https://github.com/arista-netdevops-community/anta/commit/9d55318d4f5fef3aaff7b400857ded06305b5e90

**Comment by gmuloc** [comment]
Need to fix `collect_scheduled_show_tech` as well as it creates its own set of commands..

**Review by mtache — APPROVED** [review]

**Pull request merged**